### PR TITLE
Remove /datum/lifeprocess/viruses

### PR DESCRIPTION
**Pull request by FlameArrow57**
[INTERNAL][PERFORMANCE]

## About the PR 
Removes /datum/lifeprocess/viruses and reworks it to be an event based status effect

## Why's this needed? 
Minor performance

### CODE REVIEW

**Comment by FlameArrow57** [comment]
Requires #19172 to be merged first before working properly

**Comment by FlameArrow57** [comment]
above pr is closed now

File: code/modules/status_system/statusEffects.dm [hunk 1]
```diff
@@ -2832,3 +2832,31 @@
...
+		var/mult = max(LIFE_PROCESS_TICK_SPACING, src.passed) / LIFE_PROCESS_TICK_SPACING
+		src.passed = 0
+
+		var/mob/living/L = src.owner
+
+		if (!isdead(L))
+			for (var/datum/ailment_data/ailment as anything in L.ailments)
+				ailment.stage_act(mult)
+
+		for (var/mob/living/carbon/C in oviewers(4, L))
```
Comment (ZeWaka) [inline code/modules/status_system/statusEffects.dm:2855]: you probably want hearers here?

File: code/modules/status_system/statusEffects.dm [hunk 2]
```diff
@@ -2852,9 +2852,9 @@
 			for (var/datum/ailment_data/ailment as anything in L.ailments)
 				ailment.stage_act(mult)
 
-		for (var/mob/living/carbon/C in oviewers(4, L))
-			if (prob(40))
-				L.viral_transmission(C, "Airborne", 0)
+		for (var/mob/living/other_mob in hearers(4, L))
```
Comment (ZeWaka) [inline code/modules/status_system/statusEffects.dm:2855]: makes sense also since they could be on the same time lol

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**